### PR TITLE
Update tag propagation logic

### PR DESCRIPTION
**Pull request by a-alveyblanc**
- Move tag propagation graph generation to  [pytools](https://github.com/inducer/pytools/pull/220)
- Respect `IgnoredForPropagationTag` in tag propagation

### CODE REVIEW

File: requirements.txt
Comment (a-alveyblanc) [inline requirements.txt:1]: Revert this

File: pytato/transform/metadata.py
Comment (a-alveyblanc) [inline pytato/transform/metadata.py:1]: Add example using tensor product axes

**Comment by a-alveyblanc** [comment]
Looks like pylint and mypy are failing even though the changes in https://github.com/inducer/pytools/pull/220 have been merged. They don't fail locally :man_shrugging:

**Comment by inducer** [comment]
https://pypi.org/project/pytools/2024.1.14/

**Comment by inducer** [comment]
We should add a dependency on that version in `pyproject.toml`.

**Comment by a-alveyblanc** [comment]
Added dependencies to `setup.py`. It seems like that's where other project information lives, and adding dependencies to `pyproject.toml` caused some errors.

File: pytato/transform/metadata.py [hunk 1]
```diff
@@ -687,6 +656,12 @@ def __call__(self, expr: ArrayOrNames) -> ArrayOrNames:  # type: ignore[override
         assert isinstance(result, (Array, AbstractResultWithNamedArrays))
         return result
 
+# }}}
+
+
+class IgnoredForPropagationTag(Tag):
```
Comment (inducer) [inline pytato/transform/metadata.py:662]: - Include in docs.
- Explain the semantics.
- Add a use case/justification for why one might possibly want this.

**Review by inducer — APPROVED** [review]

**Pull request merged**